receiver/clickhousesystemtablesreceiver/query_log.go

```diff
@@ -0,0 +1,361 @@
+	UsedDataTypeFamilies []string `ch:"used_data_type_families"`
+	// Canonical names of dictionaries, which were used during query execution.
+	UsedDictionaries []string `ch:"used_dictionaries"`
+	// Canonical names of formats, which were used during query execution.
+	UsedFormats []string `ch:"used_formats"`
+	// Canonical names of functions, which were used during query execution.
+	UsedFunctions []string `ch:"used_functions"`
+	// Canonical names of storages, which were used during query execution.
+	UsedStorages []string `ch:"used_storages"`
+	// Canonical names of table functions, which were used during query execution.
+	UsedTableFunctions []string `ch:"used_table_functions"`
+	UsedRowPolicies    []string `ch:"used_row_policies"`
+
+	// `transaction_id` Tuple(UInt64, UInt64, UUID),
+	TransactionId []any `ch:"transaction_id"`
+
+	// Usage of the query cache during query execution. Values: \'Unknown\' = Status unknown, \'None\' = The query result was neither written into nor read from the query cache, \'Write\' = The query result was written into the query cache, \'Read\' = The query result was read from the query cache.
+	QueryCacheUsage string `ch:"query_cache_usage"`
+
+	AsyncReadCounters map[string]uint64 `ch:"asynchronous_read_counters"`
+}
+
+func scrapeQueryLogTable(
+	ctx context.Context,
+	db driver.Conn,
+	minTs uint32,
+	maxTs uint32,
+) ([]QueryLog, error) {
+	query := fmt.Sprintf(`
+		select
+			hostname,
+			type,
+			event_date,
+			event_time,
+			event_time_microseconds,
+			query_start_time,
+			query_start_time_microseconds,
+			query_duration_ms,
+			read_rows,
+			read_bytes,
+			written_rows,
+			written_bytes,
+			result_rows,
+			result_bytes,
+			memory_usage,
+			current_database,
+			query,
+			formatted_query,
+			query_kind,
+			normalized_query_hash,
+			databases,
+			tables,
+			columns,
+			partitions,
+			projections,
+			views,
+			exception_code,
+			exception,
+			stack_trace,
+			is_initial_query,
+			user,
+			query_id,
+			address,
+			port,
+			initial_user,
+			initial_query_id,
+			initial_address,
+			initial_port,
+			initial_query_start_time,
+			initial_query_start_time_microseconds,
+			interface,
+			is_secure,
+			os_user,
+			client_hostname,
+			client_name,
+			client_revision,
+			client_version_major,
+			client_version_minor,
+			client_version_patch,
+			http_method,
+			http_user_agent,
+			http_referer,
+			forwarded_for,
+			quota_key,
+			distributed_depth,
+			revision,
+			log_comment,
+			thread_ids,
+			peak_threads_usage,
+			ProfileEvents,
+			Settings,
+			used_aggregate_functions,
+			used_aggregate_function_combinators,
+			used_database_engines,
+			used_data_type_families,
+			used_dictionaries,
+			used_formats,
+			used_functions,
+			used_storages,
+			used_table_functions,
+			used_row_policies,
+			transaction_id,
+			query_cache_usage,
+			asynchronous_read_counters,
+		from system.query_log
```

Comment:
I think this will fetch the data from only one of the nodes. You will probably have to use `clusterAllReplicas` https://clickhouse.com/docs/en/sql-reference/table-functions/cluster.